### PR TITLE
feature: implement knative service discovery

### LINKED ISSUE #402: New service discovery implementation based on Knative
We would need a new service discovery implementation in Stork for KNative. This SD locates the KNative service (and not the pods as we do for Kubernetes). It would not be possible to use load balancing. However, if you use KNative, load balancing can be handled there.

This SD would take a (Knative) service name as parameter and search for it in the current (or configured) namespace. It will not check if there are pods, just locate the service, resolve the name and return that name. It fails if no service can be found. You will never have more than one service (this needs to be verified when using revisions).

See [Quarkus #146](https://github.com/quarkusio/quarkus-super-heroes/issues/146)

### PR DESCRIPTION
**Pull request by aureamunoz**
Fixes #402

### CODE REVIEW

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/docs/service-discovery/knative.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+Supported attributes are the following:
+
+--8<-- "../service-discovery/knative/target/classes/META-INF/stork-docs/knative-sd-attributes.txt"
+
+
+## Caching the service instances
+
+Contacting the cluster too much frequently can result in performance problems. It's why Knative Service discovery extends `io.smallrye.stork.impl.CachingServiceDiscovery` to automatically _cache_ the service instances.
+Moreover, the caching expiration has been also improved in order to only update the retrieved set of `ServiceInstance` if some of them changes and an event is emitted. 
+This is done by creating an [Informer](https://www.javadoc.io/static/io.fabric8/kubernetes-client-api/6.1.1/io/fabric8/kubernetes/client/informers/SharedIndexInformer.html), similar to a [Watch](https://www.javadoc.io/static/io.fabric8/kubernetes-client-api/6.1.1/io/fabric8/kubernetes/client/Watch.html),  able to observe the events on the Knative Service instances resources.
```
[the remark this answers is not in the document]
Comment (aureamunoz) [reply docs/docs/service-discovery/knative.md:100]: You mean that Knative SD shouldn't be cacheable?

[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/docs/service-discovery/knative.md:100]: Actually, it must be cached, or we are going to do too many API calls. So ignore my comment (brain was in post-PTO mode)

[the remark this answers is not in the document]
Comment (aureamunoz) [reply docs/docs/service-discovery/knative.md:100]: No problem. Indeed, I ask myself same question when coding it. Also, I saw (if I recall correctly) that the knative service resource didn't change on different calls so it makes sense to cache it (the pod changes but not the kn-service which is the resources that stork is watching).

File: docs/docs/service-discovery/knative.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Knative Service Discovery
+
+
```
[the remark this answers is not in the document]
Comment (aureamunoz) [reply docs/docs/service-discovery/knative.md:8]: Done

**Review by cescoffier — APPROVED** [review]

**Pull request merged**